### PR TITLE
fix(web.mailpro): add dkim feature

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-11858 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added the add dkim feature

## Related
:flags: Translation:
- [ ] CMT-3110

### CODE REVIEW

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    this.services.$scope.configDkim = () => this.configDkim();
+
+    this.dkimGuideLink =
+      DKIM_CONFIGURATION_GUIDE[coreConfig.getUser().ovhSubsidiary] ||
+      DKIM_CONFIGURATION_GUIDE.DEFAULT;
+
+    this.init();
+  }
+
+  init() {
```
Comment (qpavy) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:36]:
```suggestion
  $onInit() {
```

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    this.dkimStatus = $scope.currentActionData.dkimStatus;
+    this.GLOBAL_DKIM_STATUS =
+      $scope.currentActionData.constant.GLOBAL_DKIM_STATUS;
+    this.services.$scope.configDkim = () => this.configDkim();
+
+    this.dkimGuideLink =
+      DKIM_CONFIGURATION_GUIDE[coreConfig.getUser().ovhSubsidiary] ||
+      DKIM_CONFIGURATION_GUIDE.DEFAULT;
+
+    this.init();
```
Comment (qpavy) [inline packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:33]: use $onInit to make the construction lighter

**Review by qpavy — CHANGES_REQUESTED** [review]

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:36]: In fact I need the result in the constructor the $onInit is played too late. If I do a $onInit I break the data-wizard button management.

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:33]: In fact I need the result in the constructor the $onInit is played too late. If I do a $onInit I break the data-wizard button management.

File: packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js (hunk 1 above)
Comment (qpavy) [reply packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js:36]: Ok well keep like this !

**Review by qpavy — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by acavalloni — APPROVED** [review]

File: packages/manager/modules/emailpro/src/domain/emailpro-domain.html [hunk 3]
```diff
@@ -251,6 +251,36 @@
                                         data-simplepopover-placement="top"
                                         data-simplepopover-single="true"
                                     ></button>
+
+                                    <!-- dkim -->
+                                    <button
+                                        class="oui-badge oui-badge_success"
+                                        data-translate="emailpro_tab_domain_diagnostic_dkim"
+                                        data-ng-if=":: dkimGlobalStatus(domain) === GLOBAL_DKIM_STATUS.OK && !exchange.isMXPlan"
+                                        disabled
```
Comment (anooparveti) [inline packages/manager/modules/emailpro/src/domain/emailpro-domain.html:260]: **question:** Should this button be disabled always?

Comment (ghyenne) [reply packages/manager/modules/emailpro/src/domain/emailpro-domain.html:260]: Yes if we have badge_success (if the configuration is good there isn't action onclick).

**Review by anooparveti — APPROVED** [review]

**Pull request merged**